### DISCUSSION TITLE
Seeking Permission for Creating a New Integration Project

### DISCUSSION THREAD

**Discussion by omerfaruk-aran** [omerfaruk-aran/esphome_samsung_hvac_bus#210] [Questions] [answered]
Hello @lanwin,

First and foremost, thank you for all the hard work you’ve put into this project. It has been a pleasure collaborating and contributing alongside you.

I am considering developing a new integration solution that would build upon the samsung_ac component we've been working on. The idea is to create a streamlined process where users can directly load firmware updates to their ESP devices and integrate them with Home Assistant via a dedicated module in HACS. This would allow users to set up key parameters without manually editing the example.yaml file and manage configurations through a graphical interface.

This new solution would be an independent and comprehensive integration, with updates managed through GitHub Releases using version control. Since this concept is based on the foundation you’ve established, I want to ensure that I have your permission to proceed. If agreeable to you, I would also like to acknowledge you as the initiator of this project and include your contributions as foundational.

Could you please share your thoughts on this idea? I aim to proceed with full respect and proper referencing of your work. If you're comfortable with this, I would be honored to have you involved as the original developer and project advisor.

I look forward to hearing your feedback and insights.

Thank you,
Ömer Faruk Aran

**Comment by tristone-cz** [comment]
Hi.

For sure nice project, nice thing to play 😄 

But is it worth to making the whole thing so much user friendly? At the beginning there is ESP + RS485 assembly, AC opening and connecting everything together. If someone handles that, one XML config file is the least problem 🤔 

Regards,
Jaroslav

**Comment by omerfaruk-aran** [comment]
Hello @tristone-cz ,

Thank you for your feedback and thoughts! 😊

You are right that the ESP + RS485 setup and connecting to the AC unit requires some technical knowledge. For those who can handle these tasks, an XML-based configuration file might not seem like a big obstacle. However, my aim to make this project more user-friendly is based on several key points:

Expanding Reach and User Diversity: I want this project to cater not only to technically skilled users but also to those with less technical knowledge who are interested in integrated solutions. A system that users can set up with just a few clicks would make the project more accessible.

Easier Management and Updates: With HACS integration and a GUI-based setup, managing updates will be easier. Users will be able to access new features quickly without dealing with manual file updates or changes.

Easier Support and Troubleshooting: A system that requires fewer manual configurations reduces user errors and improves support processes. Additionally, a graphical interface makes it easier to understand issues and provide feedback.

Furthermore, in the future, I plan to design a module that allows users to simply purchase a device with a setup diagram and connect it to the AC unit by plugging in a socket or screwing in two wires. Currently, since the project is C++ based, optimization and development are limited. However, as we transition to HACS, we will have more flexibility for extensive modifications using Python.

I’d love to hear more opinions on this topic.

Thank you,
Ömer Faruk Aran

**Comment by tristone-cz** [comment]
Hi.

First, I want to say, I am not against. Just elaborating if it is not wasting of time and labor on your side.
In general the idea is nice.

All your point are valid, I agree.
The question related to these users is if there is a demand for such solution. I believe that for most of low-skilled people the AC can be integrated to SmartThings and SmartThings can be integrated to HomeAssistant. The ecosystem already exists.
Not sure if exists some old NonNASA device which in parallel does not have SmartThings.
It is cloud dependent 😭 but how many local control fanatics (like eg. me) is out there?

For the upgrades, if not on daily basis, the ESPhome addon to HomeAssistant does the job already now, partially. At least every month it suggests to update to newer version of ESPhome and if there will be new version of Samsung AC it will be taken into account.
Of course, no updates of Samsung AC explicitly but, to be honest, this is supposed to be "set and forget" device.

And enough from myself. let's see some more opinions 😏

**Comment by lanwin** [comment, marked as answer]
Hi @omerfaruk-aran I like your idea. It should be much simple if the ESP simply uses the stream server and decoding and configuration is done in Python. I hate the C++ code. I can write it, but I never like it. This way it would be also possible to adapt the the REST interface some devices have. But keep in mind that there are some users witch did not use HA.

I have an offer for you. I can not put much time into this project and to be honest it really stresses me to not being able to respond to the stuff witch is going on here. And that will not change in the near future. But it seems you are able willing to invest that time witch is required by such a project. 

So would you like to take over the project? For me its enough if I got mentioned.

**Comment by omerfaruk-aran** [comment]
Hello @lanwin,

First and foremost, thank you for this generous offer and for trusting me with the contributions I’ve made to the project. It is an honor for me to accept the opportunity to take over the project, and I want to make the most of this chance. I completely align with your vision to further develop the ESPHome and Samsung HVAC Bus module in a way that broadens its reach and enhances its usability.

As you mentioned, working on a Python-based approach and creating a structure that aligns with devices beyond Home Assistant is an excellent idea. Rest assured, I will continue the project by preserving the foundations you have established, always acknowledging you as the project's initiator.

Your contributions, support, and guidance have always been invaluable. Your feedback on future developments and any project-related matters will continue to be of utmost importance to me. I look forward to continuing this journey with your support and advice.

🙏 Thank you again, and I am proud to confirm my acceptance of taking over the project!

Best regards,
Ömer Faruk Aran

**Reply by lanwin** [reply]
Can you please remove or rename your fork. Otherwise I can not transfer this project.

**Reply by omerfaruk-aran** [reply]
I have removed my fork as requested. 🎯 The path is clear for the transfer now!
